### PR TITLE
validatorapi: add node version

### PR DESCRIPTION
**Pull request by xenowits**
Adds node version endpoint to validatorapi.

category: feature
ticket: #1747

### CODE REVIEW

**Comment by xenowits** [comment]
Sample output running latest charon with compose:
Screenshot 2023-02-03 at 19 56 32

File: core/validatorapi/eth2types.go [hunk 1]
```diff
@@ -167,3 +167,13 @@ type syncCommitteeDutiesResponse struct {
...
 	Data *altair.SyncCommitteeContribution `json:"data"`
 }
+
+// nodeVersionResponse defines the response to the node version endpoint.
+// See: https://ethereum.github.io/beacon-APIs/#/Node/getNodeVersion
+type nodeVersionResponse struct {
+	Data nodeVersionInner `json:"data"`
+}
+
+type nodeVersionInner struct {
```
Comment (gsora) [inline core/validatorapi/eth2types.go:177]: Is this struct needed? Can't we merge it into `nodeVersionResponse`?

The struct is small anyway, and only used to return one value.

**Review by gsora — APPROVED** [review]

File: core/validatorapi/eth2types.go (hunk 1 above)
Comment (dB2510) [reply core/validatorapi/eth2types.go:177]: +1
Something like:
```
type nodeVersionResponse struct {
	Data struct {
		Version string `json:"version"`
	} `json:"data"`
}
```

**Review by dB2510 — APPROVED** [review]

File: core/validatorapi/router_internal_test.go [hunk 2]
```diff
@@ -962,6 +937,15 @@ func testRawRouter(t *testing.T, handler testHandler, callback func(context.Cont
 	callback(context.Background(), server.URL)
 }
 
+// newTestHandler returns a new testHandler with NodeVersionFunc set.
+func newTestHandler() testHandler {
+	return testHandler{
+		NodeVersionFunc: func(ctx context.Context) (string, error) {
```
Comment (corverroos) [inline core/validatorapi/router_internal_test.go:943]: why is this required?

File: core/validatorapi/validatorapi.go [hunk 3]
```diff
@@ -1024,6 +1026,13 @@ func (c Component) ValidatorsByPubKey(ctx context.Context, stateID string, pubsh
 	return c.convertValidators(valMap)
 }
 
+// NodeVersion returns the current version of charon.
+func (Component) NodeVersion(context.Context) (string, error) {
+	commitSHA, _ := version.GitCommit()
+
+	return fmt.Sprintf("obolnetwork/charon/%s-%s/%s-%s", version.Version, commitSHA, runtime.GOARCH, runtime.GOOS), nil
```
Comment (corverroos) [inline core/validatorapi/validatorapi.go:1033]: is `runtime.GOARCH` and `runtime.GOOS` present in the container?

File: core/validatorapi/router_internal_test.go (hunk 2 above)
Comment (xenowits) [reply core/validatorapi/router_internal_test.go:943]: Since this endpoint is required for go-eth2-client to initialize - https://github.com/attestantio/go-eth2-client/blob/master/http/service.go#L142

File: core/validatorapi/validatorapi.go (hunk 3 above)
Comment (xenowits) [reply core/validatorapi/validatorapi.go:1033]: Yes it should be. I checked it running latest charon on my PC:
Screenshot 2023-02-06 at 13 05 45

File: core/validatorapi/router_internal_test.go (hunk 2 above)
Comment (corverroos) [reply core/validatorapi/router_internal_test.go:943]: Suggest adding it a method on `testHandler` and removing the `/eth/v1/node/version` endpoint from `newBeaconHandler` since it isn't used anymore.
```
func (h testHandler) NodeVersion(ctx) string, err {
  if h.NodeVersionFunc != nil {
    return h.NodeVersionFunc(ctx)
  }
  // Default to 
  return "mock_version", nil
}
```
